documents/process/rust_versions.md

```diff
@@ -26,3 +26,116 @@ When the nightly cronjob CI fails, ICU4X developers must fix it and ensure it do
 
 
  [stability-policy]: https://rust-lang.github.io/rfcs/1122-language-semver.html
+
+
+### Pinned Rust nightly policy
+
+ICU4X pins a Rust Nightly in CI for multiple purposes:
+
+ - `diplomat-coverage/src/main.rs`: Running the diplomat-coverage task, which uses unstable Rustdoc JSON output, requiring synchronization between the Rust version and the `rustdoc_types` dependency.
+ - `PINNED_CI_NIGHTLY`: Running various FFI CI jobs that need _some_ Rust nightly to use build-std or other nightly features, without becoming beholden to arbitrary
+ - `LLVM_COMPATIBLE_NIGHTLY`: Usage of `-Clinker-plugin-lto` in size-optimized tests where Rust and Clang need to be using compatible LLVM versions.
+
+
+ ICU4X also runs its entire CI against the latest nightly once a day, reporting errors to Slack. These errors help us catch upstream breakages early, but ICU4X does not have any particular SLO against these being fixed. We guarantee we build on the latest stable as detailed in the first section of this page.
+
+ The first two nightlies here can be updated whenever necessary to whatever nightly works. It not necessary to keep the nightly versions the same, though it is ideal to try.
+
+
+ We have a stricter policy around updating `LLVM_COMPATIBLE_NIGHTLY`. We recommend usage of ICU4X with `-Clinker-plugin-lto` for size-constrained clients who have the option available, so `LLVM_COMPATIBLE_NIGHTLY` must be updated to nightlies that work with "widely available LLVM" versions. This may further constrain our MSRV policy above in case `LLVM_COMPATIBLE_NIGHTLY` cannot be pinned to something that works with an otherwise-allowed MSRV.
```

Comment:
> `LLVM_COMPATIBLE_NIGHTLY` must be updated to nightlies that work with "widely available LLVM" versions

`LLVM_COMPATIBLE_NIGHTLY` is just a CI thing, it's not something we advertise. I think this whole paragraph should be rephrased, the reasoning should go
* We recommend usage of ICU4X with `-Clinker-plugin-lto` for size-constrained clients
* Hence there needs to be a nightly that works with ICU4X and with "widely available LLVM" versions
* This CI test asserts that such a nightly exists

Reply:
Calling it LLVM_COMPATIBLE_NIGHTLY fits in better with the rest of this document: it's _what that nightly is called here_, the section starts with a list of three nightlies, giving them names. I think it's not very useful to introduce another name for this nightly. I'll reframe to make it more like a guarantee, but we need to have a name for it, and I think the env var is the best choice.

Reply:
I've reworded it to focus on the guarantee. I initially intended this to be an internal doc, but it is part of our external guarantees, so tried to focus on those more.